### PR TITLE
feat: Add Social cards #1347

### LINKED ISSUE #1347: Feature: Add social card
### Type of feature

Feature

### Current behavior

When sharing this site on social, there is no social card

### Desired solution

Set up a social card with the following image. 

social=image

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by LevinGar**
## What type of PR is this?
- [x] ✨ Feature

## Description
This PR adds Social Cards as requested in #1347 by implementing Open Graph Protocol meta tags and twitter tags.

## Related Tickets & Documents
Resolves #1347 
https://www.npmjs.com/package/react-helmet-async

## Mobile & Desktop Screenshots/Recordings
A demonstration of how the Social Cards look in different platforms:
Screen Shot 2021-12-23 at 8 29 01 p m

## Added tests?
- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by mtfoley** [comment]
Hi @LevinGar thank you for working on this one! My comments are actually directed at other maintainers since I don’t have a lot of experience with react/helmet.

- @bdougie / @0-vortex is there a reason we wouldn’t have these meta tags just be part of markup in index.html without the use of helmet? 
- Also, if the package.json is modified the npm-shrinkwrap.json ought to be included also, correct?

**Comment by 0-vortex** [comment]
> Hi @LevinGar thank you for working on this one! My comments are actually directed at other maintainers since I don’t have a lot of experience with react/helmet.
> 
> * @bdougie / @0-vortex is there a reason we wouldn’t have these meta tags just be part of markup in index.html without the use of helmet?
> * Also, if the package.json is modified the npm-shrinkwrap.json ought to be included also, correct?

That is correct about the dependencies, however, I don't see a point for async helmet. Didn't think about having it in plain HTML either

**Comment by bdougie** [comment]
I am familiar with react helmet and have used it. The biggest benefit is the use of dynamic social cards.

For example if we wanted to share opensauced links for specific repos (opensauced.pizza/repos/Facebook/react), a social card can be generated. Making this static in index.html would make it less flexible. 

I am fine to include it for that reason and that it is a common dependency in react projects.

**Comment by 0-vortex** [comment]
> I am familiar with react helmet and have used it. The biggest benefit is the use of dynamic social cards.
> 
> I am fine to include it for that reason and that it is a common dependency in react projects.

My concern is with the async fork of it in a PWA, we could even innovate a bit into not needing SSR for the cards part, pre-rendering, serverless, lots of options 🍕

**Comment by LevinGar** [comment]
During the development, I originally implemented Helmet without the async fork, but it was dropping a warning with componentWillMount that is not that compatible with the latest react versions, additionally, I did some research and found other projects with the same problem, they solved it with the async as in this case: https://github.com/nfl/react-helmet/issues/623. There is no major difference with regular Helmet, the only difference is the implementation of HelmetProvider.
In case it is needed to implement the meta tags dynamically in a near future, you can assign the component title and description of MetaDecorator.

File: src/components/MetaDecorator.js
Comment (bdougie) [inline src/components/MetaDecorator.js:18]: Can we replace the URL here to "https://repository-images.githubusercontent.com/71359796/179eca00-061e-11eb-9b57-856a179db1a1" it matches the uploaded social for the repo

File: src/components/MetaDecorator.js
Comment (bdougie) [inline src/components/MetaDecorator.js:26]: replace url https://repository-images.githubusercontent.com/71359796/179eca00-061e-11eb-9b57-856a179db1a1

**Review by bdougie — CHANGES_REQUESTED** [review]
One small fix for the URL.

**Comment by bdougie** [comment]
Thanks for the context @LevinGar. I think there is enough here to merge, I just left a review for the URL to be updated to the link to the github repo social card image.

**Review by bdougie — APPROVED** [review]
LGTM

**Comment by 0-vortex** [comment]
The comments could go in as well IMHO!

**Comment by bdougie** [comment]
@LevinGar can you merge upstream with no ff. There are conflicts and some vite updates that we will need before merging this. Thanks

**Comment by 0-vortex** [comment]
In order to hydrate `react-helmet` we need to apply an SSR script similar to https://github.com/vitejs/vite/blob/main/packages/playground/ssr-vue/prerender.js - the code in this pull request won't work without it as we are serving a static build.

Another way to fast forward this pull request is to add the social card statically inside `index.html` and open another ticket for static rendering - like that we can merge all the code in this PR and continue work on top of the existing `react-helmet` changes at a later stage! 🍕

**Comment by bdougie** [comment]
Going to close. Found some issues with the Helmet being JS and not JSX. Also needs to work with vite now as well.

**Pull request closed without merging**
